iotedgehubdev/errors.py

```diff
@@ -39,29 +39,26 @@ def __init__(self, msg, ex=None):
         super(EdgeDeploymentError, self).__init__(msg, ex)
 
 
-class ResponseError(Exception):
+class ResponseError(EdgeError):
     def __init__(self, status_code, value):
+        super(ResponseError, self).__init__(value)
         self.value = value
         self.status_code = status_code
 
-    def message(self):
-        return ('Code:{0}. Detail:{1}').format(self.status_code, self.value)
 
-    def status(self):
-        return self.status_code
-
-
-class RegistriesLoginError(Exception):
+class RegistriesLoginError(EdgeError):
     def __init__(self, registries, errmsg):
+        super(RegistriesLoginError, self).__init__(errmsg)
         self._registries = registries
         self._errmsg = errmsg
 
-    def message(self):
+    def getmsg(self):
         return ('Fail to login {0}. Detail: {1}').format(self._registries, self._errmsg)
 
     def registries(self):
         return self._registries
 
 
-class InvalidConfigError(Exception):
-    pass
+class InvalidConfigError(EdgeError):
+    def __init__(self, msg, ex=None):
+        super(InvalidConfigError, self).__init__(msg)
```

Comment:
ex is passed in the constructor, why not pass down to the super? Besides, where do we use InvalidConfigError? Seems the init method has been changed, do we need also update there?

Reply:
ex is not used at all, we can remove it. Only one place raise this error, it would work fine:
```
raise InvalidConfigError(str(e))
```

Reply:
If we don't want user to init InvalidConfigError, remove the ex=None in init method?

Reply:
Updated